src/time_tracker/auth/core.clj

```diff
@@ -1,37 +1,52 @@
 (ns time-tracker.auth.core
-  (:gen-class)
   (:require [ring.util.response :as res]
             [org.httpkit.client :as http]
-            [cheshire.core :as cheshire]))
+            [cheshire.core :as json]
 
-(defn is-valid?
-  "Validates a JWT token by calling Google's API and by checking the client ID."
+            [time-tracker.config :as config]))
+
+(defn- munge-keyword
+  "In: \"key_string\"
+  Out: :key-string"
+  [key-string]
+  (keyword (clojure.string/replace key-string #"_" "-")))
+
+(defn- call-google-tokeninfo-api
+  [token]
+  (let [{body :body :as response} @(http/get config/google-tokeninfo-url
+                                             {:as :text
+                                              :query-params {"id_token" token}})]
+    (assoc response :body (json/parse-string body munge-keyword))))
+
+(defn token->credentials
+  "Validates a JWT by calling Google's API and by checking the client ID."
   [client-ids token]
-  (let [url (str "https://www.googleapis.com/oauth2/v3/tokeninfo?id_token="
-                 token)
-        response (http/get url {:as :text})]
-    (if (= 200 (:status @response))
-      (let [response-data (cheshire/parse-string (:body @response)
-                                                 #(keyword (clojure.string/replace % #"_" "-")))]
-        (and (some #{(:aud response-data)} client-ids)
-             response-data)))))
+  (let [{:keys [status body]} (call-google-tokeninfo-api token)]
+    (if (and (= 200 status)
+             (some #{(:aud body)} client-ids))
+      body)))
+
+(defn token-from-headers
+  "Extracts the token from a Ring header map, if present.
+  See: https://tools.ietf.org/html/rfc6750#section-2.1"
+  [ring-headers]
+  (if-let [header-value (get ring-headers "authorization")]
+    (let [[scheme token] (clojure.string/split header-value #" ")]
```

Comment:
That's the RFC in which the bearer token scheme for OAuth is described. Now that I think of it though, I think that [this link](https://jwt.io/introduction/) is much more relevant. I'll change this.